flocker/node/agents/ebs.py

```diff
@@ -224,4 +409,24 @@ def destroy_volume(self, blockdevice_id):
         raise UnknownVolume(blockdevice_id)
 
     def get_device_path(self, blockdevice_id):
-        pass
+        """
+        Get device path for the EBS volume corresponding to the given
+        block device.
+
+        :param unicode blockdevice_id: EBS UUID for the volume to look up.
+
+        :returns: A ``FilePath`` for the device.
+        :raises UnknownVolume: If the supplied ``blockdevice_id`` does not
+            exist.
+        :raises UnattachedVolume: If the supplied ``blockdevice_id`` is
+            not attached to a host.
+        """
+        volume = self._get(blockdevice_id)
+        if volume.attached_to is None:
+            raise UnattachedVolume(blockdevice_id)
+
+        ebs_volume = self._get_ebs_volume(blockdevice_id)
+        device = ebs_volume.tags[ATTACHED_DEVICE_LABEL]
+        if device is None:
+            raise UnattachedVolume(blockdevice_id)
```

Comment:
There's already a check for the volume being unattached above.  So this case never gets executed by the test suite.  I think it may not work, either, because `ebs_volume.tags[ATTACHED_DEVICE_LABEL]` probably raises a `KeyError` if that tag doesn't exist (eg because it was deleted by an earlier call to `detach_volume`).  I don't think this case is impossible - someone could modify the tags in between the `self._get(...)` call and the `self._get_ebs_volume(...)` call - but our test suite doesn't exercise it.

If you can think of a way to test this off the top of your head, add a test that demonstrates this works (or that demonstrates it needs to handle `KeyError` and then fix the implementation).  If it's hard to test, I suggest removing this second bit of code handling the unattached case (only this second bit, the first bit above is good) and filing an issue for addressing the more obscure failure case later (and link to it from here).